Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TrackModifier.cpp

```diff
@@ -0,0 +1,118 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/TruthTracking/TrackModifier.hpp"
+
+#include "Acts/EventData/MultiTrajectory.hpp"
+#include "Acts/Utilities/ThrowAssert.hpp"
+#include "ActsExamples/EventData/Track.hpp"
+#include "ActsExamples/EventData/Trajectories.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+
+#include <cmath>
+#include <cstdint>
+#include <stdexcept>
+#include <vector>
+
+ActsExamples::TrackModifier::TrackModifier(const Config& config,
+                                           Acts::Logging::Level level)
+    : BareAlgorithm("TrackModifier", level), m_cfg(config) {
+  if (m_cfg.inputTrajectories.empty() == m_cfg.inputTrackParameters.empty()) {
+    throw std::invalid_argument(
+        "Exactly one of trajectories or track parameters input must be set");
+  }
+  if (m_cfg.outputTrajectories.empty() == m_cfg.outputTrackParameters.empty()) {
+    throw std::invalid_argument(
+        "Exactly one of trajectories or track parameters output must be set");
+  }
+  if (m_cfg.inputTrajectories.empty() != m_cfg.outputTrajectories.empty()) {
+    throw std::invalid_argument(
+        "Input and output for trajectories and track parameters have to be "
+        "used consistently");
+  }
+}
+
+ActsExamples::ProcessCode ActsExamples::TrackModifier::execute(
+    const ActsExamples::AlgorithmContext& ctx) const {
+  auto modifyTrack = [this](auto trk) {
```

Comment:
I think we should decide if we make a copy in the function parameter, or explicit in the line below